### PR TITLE
Implemented list-dead-letter-source-queues API

### PR DESCRIPTION
**Pull request by BisratYalew**
**Please refer to the contribution guidelines in the README when submitting PRs.**

### CODE REVIEW

File: localstack/services/sqs/sqs_listener.py [hunk 1]
```diff
@@ -180,6 +180,32 @@ def _add_queue_attributes(path, req_data, content_str, headers):
     return content_str
 
 
+def _list_dead_letter_source_queues(queues, queue_url):
+    dead_letter_source_queues = []
+    for k, v in queues.items():
+        for i, j in v.items():
+            if(i == 'RedrivePolicy'):
```
Comment (whummer) [inline localstack/services/sqs/sqs_listener.py]: nitpick: we can remove the braces here (same in line 190):
```
if i == 'RedrivePolicy':
    ...
```

File: localstack/services/sqs/sqs_listener.py [hunk 2]
```diff
@@ -200,6 +226,19 @@ def forward_request(self, method, path, data, headers):
             elif action == 'DeleteQueue':
                 QUEUE_ATTRIBUTES.pop(_queue_url(path, req_data, headers), None)
 
+            elif action == 'ListDeadLetterSourceQueues':
+                queue_url = _queue_url(path, req_data, headers)
+                headers = {'content-type': 'application/xhtml+xml'}
+                content_str = _list_dead_letter_source_queues(QUEUE_ATTRIBUTES, queue_url)
+
+                new_response = Response()
```
Comment (whummer) [inline localstack/services/sqs/sqs_listener.py]: we can simplify lines 234-238 by adding an import at the top of the file and using `requests_response`:
```
from localstack.utils.aws.aws_responses import requests_response
...

return requests_response(content_str, headers=headers)
```

File: tests/integration/test_sqs.py [hunk 3]
```diff
@@ -275,3 +275,23 @@ def test_set_unsupported_attributes(self):
...
+        dlq = self.client.create_queue(QueueName=dlq_name)
+
+        dlq_arn = aws_stack.sqs_queue_arn(dlq_name)
+
+        attributes = {'RedrivePolicy': json.dumps({'deadLetterTargetArn': dlq_arn, 'maxReceiveCount': 100})}
+        nq = self.client.create_queue(QueueName=normal_queue_name, Attributes=attributes)['QueueUrl']
+
+        res = self.client.list_dead_letter_source_queues(QueueUrl=dlq['QueueUrl'])
+
+        self.assertEqual(res['queueUrls'][0], nq)
```
Comment (whummer) [inline tests/integration/test_sqs.py]: Perhaps add an assertion here that one dead letter source queue is returned:
```
self.assertEqual(len(res['queueUrls']), 1)
```

Comment (BisratYalew) [reply tests/integration/test_sqs.py]: I have fixed it.

File: localstack/services/sqs/sqs_listener.py [hunk 4]
```diff
@@ -180,6 +181,32 @@ def _add_queue_attributes(path, req_data, content_str, headers):
...
 
 
+def _list_dead_letter_source_queues(queues, queue_url):
+    dead_letter_source_queues = []
+    for k, v in queues.items():
+        for i, j in v.items():
+            if i == 'RedrivePolicy':
+                f = json.loads(v[i])
+                queue_url_split = queue_url.split('/')
+                if(queue_url_split[len(queue_url_split) - 1] in f['deadLetterTargetArn']):
```
Comment (whummer) [inline localstack/services/sqs/sqs_listener.py]: nit: could be simplified to
```
if queue_url_split[-1] in f['deadLetterTargetArn']:
    ...
```

File: tests/integration/test_sqs.py [hunk 5]
```diff
@@ -275,3 +275,24 @@ def test_set_unsupported_attributes(self):
...
+        self.assertEqual(res['ResponseMetadata']['HTTPStatusCode'], 200)
+
+        # clean up
+        self.client.delete_queue(QueueUrl=nq)
+        self.assertEqual(len(res['queueUrls']), 1)
```
Comment (whummer) [inline tests/integration/test_sqs.py]: nit: shouldn't this be moved up to line 292 (before `self.assertEqual(res['queueUrls'][0], nq)`)?

Not a bit deal, but technically this assertion is not part of the "clean up" logic.

**Comment by whummer** [comment]
LGTM, thanks @BisratYalew 👍

**Pull request merged**